public/pages/Rules/components/RuleEditor/DetectionVisualEditor.tsx

```diff
@@ -212,11 +250,15 @@ export class DetectionVisualEditor extends React.Component<
     };
 
     selections.forEach((selection) => {
-      const selectionMaps: any = {};
+      let selectionMaps: any = {};
 
       selection.data.forEach((datum) => {
-        const key = `${datum.field}${datum.modifier ? `|${datum.modifier}` : ''}`;
-        selectionMaps[key] = datum.values;
+        if (datum.field) {
```

Comment:
Just to confirm, are there any situations where `field` might be number type `0`? If so, I think this check would return `false`, which may not be intended depending on the use here.